core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java

```diff
@@ -732,11 +732,21 @@ public static long cleanUpDeltaFiles(CarbonTable table, boolean isDryRun) throws
                 .collect(Collectors.toList()));
       }
       SegmentUpdateDetails[] updateDetails = updateStatusManager.readLoadMetadata();
-      for (SegmentUpdateDetails block : updateDetails) {
-        totalDeltaFiles.stream().filter(fileName -> fileName.getName().endsWith(block
-                .getDeleteDeltaStartTimestamp() + CarbonCommonConstants.DELETE_DELTA_FILE_EXT))
-                .collect(Collectors.toList()).forEach(fileName -> totalDeltaFiles.remove(fileName));
-      }
+
+      // Case 1: When deleteDeltaStartTimestamp = deleteDeltaEndTimestamp. in this case only 1
+      // delta file is present and deltaFileStamps is NULL
+      // Case 2: When deleteDeltaStartTimestamp != deleteDeltaEndTimestamp. in this case more
+      // than 1 delta files are present, then can blindly read deltaFilesStamps variable
+      Arrays.stream(updateDetails).forEach(block -> {
+        if (block.getDeleteDeltaStartTimestamp().equals(block.getDeleteDeltaEndTimestamp())) {
+          totalDeltaFiles.removeIf(filter -> filter.getName().endsWith(block
```

Comment:
rename `filter` to `deltaFile` or `carbonFile`

Reply:
done